test-network-function/networking/suite_test.go

```diff
@@ -0,0 +1,161 @@
+		expectedRes           string
+	}{
+		{
+			inputRes:              `[\n  {\n    \"containerPort\": 8080,\n    \"name\": \"http-probe\",\n    \"protocol\": \"UDP\"\n  }\n]`,
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{{port: 8080, protocol: "UDP"}: "http-probe"},
+			expectedRes:           `[\n  {\n    \"containerPort\": 8080,\n    \"name\": \"http-probe\",\n    \"protocol\": \"UDP\"\n  }\n]`,
+		},
+		{
+			inputRes:              "",
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{},
+			expectedRes:           "",
+		},
+		{
+			inputRes:              `[\n  {\n    \"containerPort\": 8080,\n    \"name\": \"http-probe\",\n    \"protocol\": \"TCP\"\n  }\n]`,
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{{port: 8080, protocol: "TCP"}: "http-probe"},
+			expectedRes:           `[\n  {\n    \"containerPort\": 8080,\n    \"name\": \"http-probe\",\n    \"protocol\": \"TCP\"\n  }\n]`,
+		},
+	}
+
+	for _, tc := range testCases {
+		err := parseVariables(tc.inputRes, tc.declaredPorts)
+		assert.Nil(t, err)
+		assert.Equal(t, tc.expectedDeclaredPorts, tc.declaredPorts)
+	}
+}
+
+func TestDeclaredPortList(t *testing.T) {
+	// expected inputs
+	testCases := []struct {
+		// inputs
+		container     int
+		podName       string
+		podNamespace  string
+		declaredPorts map[key]string
+
+		// expected outputs here
+		expectedDeclaredPorts map[key]string
+	}{
+		{
+			container:             0,
+			podName:               "test-54bc4c6d7-2gwlz",
+			podNamespace:          "tnf",
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{{port: 8080, protocol: "TCP"}: "http-probe"},
+		},
+		{
+			container:             0,
+			podName:               "",
+			podNamespace:          "",
+			declaredPorts:         map[key]string{},
+			expectedDeclaredPorts: map[key]string{},
+		},
+	}
+
+	for _, tc := range testCases {
+		err := declaredPortList(tc.container, tc.podName, tc.podNamespace, tc.declaredPorts)
+		assert.Nil(t, err)
+		assert.Equal(t, tc.expectedDeclaredPorts, tc.declaredPorts)
+	}
+}
+
+func TestListeningPortList(t *testing.T) {
+	// expected inputs
+	testCases := []struct {
+		// inputs
+		commandlisten  []string
+		nodeOc         *interactive.Context
+		listeningPorts map[key]string
+
+		// expected outputs here
+		expectedlisteningPorts map[key]string
+	}{
+		{
+			commandlisten:          []string{"nsenter -t 4380 -n", "ss -tulwnH"},
+			nodeOc:                 nil,
+			listeningPorts:         map[key]string{},
+			expectedlisteningPorts: map[key]string{{port: 8080, protocol: "TCP"}: ""},
+		},
+		{
+			commandlisten:          []string{},
+			nodeOc:                 nil,
+			listeningPorts:         map[key]string{},
+			expectedlisteningPorts: map[key]string{},
+		},
+	}
+	for _, tc := range testCases {
+		err := listeningPortList(tc.commandlisten, tc.nodeOc, tc.listeningPorts)
```

Comment:
Pretty much the same as the previous UT.

Reply:
I fixed it.